homeassistant/components/homekit_controller/connection.py

```diff
@@ -791,18 +790,18 @@ def _add_new_entities(self, callbacks: list[AddServiceCb]) -> None:
                         self.entities.add(entity_key)
                         break
 
-    async def async_load_platform(self, platform: str) -> None:
-        """Load a single platform idempotently."""
-        if platform in self.platforms:
+    async def _async_load_platforms(self, platforms: set[str]) -> None:
+        """Load a group of platforms."""
+        if not (to_load := platforms - self.platforms):
             return
 
-        self.platforms.add(platform)
+        self.platforms.update(to_load)
         try:
-            await self.hass.config_entries.async_forward_entry_setup(
-                self.config_entry, platform
+            await self.hass.config_entries.async_forward_entry_setups(
+                self.config_entry, platforms
             )
         except Exception:
-            self.platforms.remove(platform)
+            self.platforms -= to_load
```

Comment:
This seems kinda we shouldn't remove them since there is no reason to try loading them again as its just as likely to fail again